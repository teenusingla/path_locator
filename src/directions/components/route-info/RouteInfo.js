import React from "react";

import "./RouteInfo.scss";

/**
 * @name RouteInfoItem
 * @type {Component}
 * @description Route info item component
 * @param {{label,value}} Object Props to stateless component
 * @returns {JSX}
 */
const RouteInfoItem = ({ label, value }) => (
  <div className="route-info-item">
    <div className="route-info-item-label">{label}: </div>
    <div className="route-info-item-value"> {value}</div>
  </div>
);

/**
 * @name RouteInfo
 * @type {Component}
 * @description Route info component. Display the route params like distance and time
 * @param {{total_distance,total_time}} Object Props to stateless component
 * @returns {JSX}
 */
const RouteInfo = ({ total_distance, total_time }) => (
  <div className="route-info">
    <RouteInfoItem label="Distance" value={total_distance} />
    <RouteInfoItem label="Time" value={total_time} />
  </div>
);

export default RouteInfo;
