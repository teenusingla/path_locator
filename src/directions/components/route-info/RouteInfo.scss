.route-info {
  padding-top: 16px;
  display: flex;
}

.route-info-item {
  display: flex;
  font-size: 1em;

  &:not(:last-child) {
    margin-right: 16px;
  }
}

.route-info-item-label {
  font-weight: 500;
}

.route-info-item-value {
  margin-left: 8px;
  font-weight: bold;
}
