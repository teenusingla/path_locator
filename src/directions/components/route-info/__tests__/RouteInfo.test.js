import React from "react";
import renderer from "react-test-renderer";
import RouteInfo from "../RouteInfo";

describe("<RouteInfo /> component", () => {
  it("should match the snapshot", () => {
    const tree = renderer.create(<RouteInfo />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
