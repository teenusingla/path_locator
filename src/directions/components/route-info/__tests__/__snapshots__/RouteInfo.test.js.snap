// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RouteInfo /> component should match the snapshot 1`] = `
<div
  className="route-info"
>
  <div
    className="route-info-item"
  >
    <div
      className="route-info-item-label"
    >
      Distance
      : 
    </div>
    <div
      className="route-info-item-value"
    >
       
    </div>
  </div>
  <div
    className="route-info-item"
  >
    <div
      className="route-info-item-label"
    >
      Time
      : 
    </div>
    <div
      className="route-info-item-value"
    >
       
    </div>
  </div>
</div>
`;
